var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -258,12 +258,14 @@ class OpenfoamCom(Package):
     """
 
     homepage = "http://www.openfoam.com/"
-    baseurl  = "https://sourceforge.net/projects/openfoamplus/files"
+    baseurl  = "https://sourceforge.net/projects/openfoamplus/files/"
+    gitrepo =  "https://develop.openfoam.com/Development/OpenFOAM-plus.git"
 
+    version('1706', '630d30770f7b54d6809efbf94b7d7c8f',
+            url=baseurl + 'v1706/OpenFOAM-v1706.tgz')
     version('1612', 'ca02c491369150ab127cbb88ec60fbdf',
-            url=baseurl + '/v1612+/OpenFOAM-v1612+.tgz')
-    version('plus', branch='develop',  # Note: needs user credentials
-            git='https://develop.openfoam.com/Development/OpenFOAM-plus.git')
+            url=baseurl + 'v1612+/OpenFOAM-v1612+.tgz')
+    version('plus', branch='develop', git=gitrepo)  # Needs credentials
```

Comment:
It would be better to call this version `develop`. When Spack sorts versions, numeric versions always sort before non-numeric versions. We make an exception for `develop`, which always sorts greater than anything else.